tensorly/decomposition/CMTF_ALS

```diff
@@ -0,0 +1,165 @@
+    V : tl.tensor
+        true matrix for generating Y
+    A_hat : tl.tensor
+        estimated 1st factor matrix of tensor X
+    B_hat : tl.tensor
+        estimated 2nd factor matrix of tensor X
+    C_hat : tl.tensor
+        estimated 3rd factor matrix of tensor X
+    V_hat : tl.tensor
+        estimated matrix for generating Y
+
+    Returns
+    -------
+    factor match score : float
+        value between 0 and 1 that shows how good the estimation is, 1 is perfect
+
+    """
+
+    R = A.shape[1]
+
+    # calculate weights
+    lambda_ = tl.zeros(R)
+    lambda_hat = tl.zeros(R)
+    alpha_ = tl.zeros(R)
+    alpha_hat = tl.zeros(R)
+    xi = tl.zeros(R)
+    xi_hat = tl.zeros(R)
+    for r in range(R):
+        lambda_[r] = tl.norm(A[:, r]) * tl.norm(B[:, r]) * tl.norm(C[:, r])
+        lambda_hat[r] = tl.norm(A_hat[:, r]) * tl.norm(B_hat[:, r]) * tl.norm(C_hat[:, r])
+        alpha_[r] = tl.norm(A[:, r]) * tl.norm(V[:, r])
+        alpha_hat[r] = tl.norm(A_hat[:, r]) * tl.norm(V_hat[:, r])
+        xi[r] = lambda_[r] + alpha_[r]
+        xi_hat[r] = lambda_hat[r] + alpha_hat[r]
+
+    # take care of permutation of normalized matrices (not mentioned in paper, but necessary)
+    perm = (A.T @ A_hat) * (B.T @ B_hat) * (C.T @ C_hat) / (lambda_[:, None] @ lambda_hat[None, :])
+    order = tl.argmax(perm, axis=1)
+    perm = perm[:, order]  # perm[r,r] now corresponds to \prod_i (a^i_r.T @ ahat^i_r) / (||(a^i_r|| ||ahat^i_r||)
+    V_hat = V_hat[:, order]
+    xi_hat = xi_hat[order]
+
+    # calculate factor match score
+    FMS = tl.zeros(R)
+    for r in range(R):
+        FMS[r] = (1 - tl.abs(xi[r] - xi_hat[r]) / tl.max([xi[r], xi_hat[r]])) * tl.abs(
+            perm[r, r] * tl.dot(V[:, r], V_hat[:, r])) / (tl.norm(V[:, r]) * tl.norm(V_hat[:, r]))
+    return tl.min(FMS)
+
+
+def cmtf_als_for_third_order_tensor(X, Y, R):
+    """
+    Calculates a coupled matrix and tensor factorization of 3rd order tensor and matrix which are coupled in first mode.
+
+    Assume you have 3rd order tensor X = [[A,B,C]] and matrix Y = A @ V.T, which are coupled in 1st mode.
+    With coupled matrix and tensor factorization (CTMF), the factor matrices A, B, C for the CP decomposition of X
+    and the matrix V are found.
+    This implementation only works for a coupling in the first mode.
+
+    Solution is found via alternating least squares (ALS) as described in Figure 5 of
+    @article{acar2011all,
+      title={All-at-once optimization for coupled matrix and tensor factorizations},
+      author={Acar, Evrim and Kolda, Tamara G and Dunlavy, Daniel M},
+      journal={arXiv preprint arXiv:1105.3422},
+      year={2011}
+    }
+
+    Parameters
+    ----------
+    X : tl.tensor
+        3rd order tensor with factor matrices [[A, B, C]]
+    Y : tl.tensor
+        matrix that is coupled with tensor in first mode: Y = A @ V.T
+    R : int
+        rank for CP decomposition of X
+
+    Returns
+    -------
+    A, B, C, V : estimated matrices
```

Comment:
Could be kruskal_tensor, matrix? 
Or even we return two kruskal tensors, one third order, the other second order, with the same factor first?